extensions/panache/panache-common/deployment/src/main/java/io/quarkus/panache/common/deployment/visitors/KotlinPanacheClassOperationGenerationVisitor.java

```diff
@@ -541,8 +542,9 @@ public MethodVisitor visitMethod(int access, String name, String descriptor, Str
         } else if (name.contains("$")) {
             //some agents such as jacoco add new methods, they generally have $ in the name
             return super.visitMethod(access, name, descriptor, signature, exceptions);
-        } else if (name.equals(CTOR_METHOD_NAME)) {
+        } else if (name.equals(CTOR_METHOD_NAME) || name.equals(CLINIT_METHOD_NAME)) {
             //Arc can add no-args constructors to support intercepted beans
+            // Logging with Panache can add a class initializer
             return super.visitMethod(access, name, descriptor, signature, exceptions);
         }
         return null;
```

Comment:
Hi @evanchooly, I especially wanted to ask you about this method. It seems it almost always forwards to `super.visitMethod`, except of some cases which I don't really understand. Does this maybe just "skip" methods that are "defined" (present in `definedMethods`) and do have the `@GenerateBridge` annotation? In that case, perhaps this method should look like this?

```java
    @Override
    public MethodVisitor visitMethod(int access, String name, String descriptor, String signature,
            String[] exceptions) {
        MethodInfo methodInfo = definedMethods.entrySet().stream()
                .filter(e -> e.getKey().equals(name + descriptor))
                .map(e -> e.getValue())
                .findFirst()
                .orElse(null);
        if (methodInfo != null && methodInfo.hasAnnotation(DOTNAME_GENERATE_BRIDGE)) {
            return null;
        }
        return super.visitMethod(access, name, descriptor, signature, exceptions);
    }
```